### PR TITLE
adding messages eventhub to io-messages infra

### PR DESCRIPTION
**Pull request by lucamancosu**
#### List of Changes

Added a messages eventhub module to the infra

#### Motivation and Context

Creation of a new eventhub for broadcast messages

#### Types of changes

- [x] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: infra/resources/_modules/messages_eventhub/messages_eventhub.tf [hunk 1]
```diff
@@ -0,0 +1,27 @@
+module "messages-evh" {
+  source = "github.com/pagopa/dx//infra/modules/azure_event_hub?ref=3f205e62474782678a563f4cff92e479a34feecd"
+
+  environment = {
+    prefix          = var.prefix
+    env_short       = var.env_short
+    location        = var.location
+    domain          = var.domain
+    app_name        = "elt"
```
Comment (Garma00) [inline infra/resources/_modules/messages_eventhub/messages_eventhub.tf]:
```suggestion
    app_name        = "etl"
```

**Review by Garma00 — CHANGES_REQUESTED** [review]

File: infra/resources/_modules/messages_eventhub/messages_eventhub.tf [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    domain          = var.domain
+    app_name        = "etl"
+    instance_number = var.instance_number
+  }
+
+  eventhubs = [{
+    name                   = local.eventhub_name
+    partitions             = local.eventhub_partition_count
+    message_retention_days = local.eventhub_message_retention
+    keys                   = local.eventhub_auth_messages_keys
```
Comment (lucacavallaro) [inline infra/resources/_modules/messages_eventhub/messages_eventhub.tf]: there is no need to use `locals` here - just put the values directly

File: infra/resources/_modules/messages_eventhub/locals.tf [hunk 3]
```diff
@@ -0,0 +1,18 @@
+locals {
+  eventhub_name              = "messages-evh"
+  eventhub_partition_count   = 32
+  eventhub_message_retention = 7
+  resource_group_name        = "io-p-itn-com-rg-01"
```
Comment (lucacavallaro) [inline infra/resources/_modules/messages_eventhub/locals.tf]: is this local used?

**Review by lucacavallaro — CHANGES_REQUESTED** [review]
@lucamancosu rember to call `pre-commit run -a` before pushing infra code

File: infra/resources/_modules/messages_eventhub/messages_eventhub.tf (hunk 2 above)
Comment (lucamancosu) [reply infra/resources/_modules/messages_eventhub/messages_eventhub.tf]: Ok

File: infra/resources/_modules/messages_eventhub/locals.tf (hunk 3 above)
Comment (lucamancosu) [reply infra/resources/_modules/messages_eventhub/locals.tf]: no more needed

**Review by Garma00 — APPROVED** [review]
LGTM

File: infra/resources/prod/main.tf [hunk 4]
```diff
@@ -219,3 +219,18 @@ module "monitoring" {
   resource_group_name = azurerm_resource_group.itn_messages.name
   io_com_slack_email  = data.azurerm_key_vault_secret.io_com_slack_email.value
 }
+
+module "messages_eventhub" {
```
Comment (lucacavallaro) [inline infra/resources/prod/main.tf]: please move this block into `eventhubs.tf` folder

can you also rename the folder into `_modules` to `eventhubs`?

**Review by lucacavallaro — CHANGES_REQUESTED** [review]

File: infra/resources/prod/main.tf (hunk 4 above)
Comment (lucamancosu) [reply infra/resources/prod/main.tf]: **please move this block into eventhubs.tf folde**r -> do you mean to create a infra/resources/prod/eventhubs.tf **file** ? in this case there is no problem

If you mean to move the block inside the eventhubs folder(after renaming infra/resources/_modules/messages_eventhubfolder into infra/resources/_modules/eventhubs) we lose all the shared locals and data that are inside the infra/resources/prod folder.

Make sense ?

Comment (lucacavallaro) [reply infra/resources/prod/main.tf]: sorry, I meant file 🤦

File: infra/resources/_modules/messages_eventhub/messages_eventhub.tf [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    prefix          = var.prefix
+    env_short       = var.env_short
+    location        = var.location
+    domain          = var.domain
+    app_name        = "etl"
+    instance_number = var.instance_number
+  }
+
+  eventhubs = [{
+    name                   = "messages-evh"
```
Comment (lucacavallaro) [inline infra/resources/_modules/messages_eventhub/messages_eventhub.tf]:
```suggestion
    name                   = "io-p-com-etl-messages-evh-01"
```

(I think that `io-p-com-etl` can be obtained from variables)

File: infra/resources/_modules/messages_eventhub/messages_eventhub.tf [hunk 6]
```diff
@@ -0,0 +1,38 @@
+module "messages-evh" {
+  source = "github.com/pagopa/dx//infra/modules/azure_event_hub?ref=3f205e62474782678a563f4cff92e479a34feecd"
+
+  environment = {
+    prefix          = var.prefix
+    env_short       = var.env_short
+    location        = var.location
+    domain          = var.domain
+    app_name        = "etl"
+    instance_number = var.instance_number
```
Comment (lucacavallaro) [inline infra/resources/_modules/messages_eventhub/messages_eventhub.tf]: no var needed here :) 

```suggestion
    instance_number = "01"
```

**Review by lucacavallaro — CHANGES_REQUESTED** [review]
please rename/create the following file/folders

* `_modules/messages_eventhub` -> `_modules/eventhubs`
* `_modules/messages_eventhub` -> `_modules/eventhubs/etl.tf`
* `eventhubs.tf`

File: infra/resources/_modules/messages_eventhub/messages_eventhub.tf (hunk 5 above)
Comment (lucamancosu) [reply infra/resources/_modules/messages_eventhub/messages_eventhub.tf]: the eventhub name is composed by the eventhub dx module. As you can see from the terraform plan the entire name is **io-p-itn-com-etl-messages-evh-01**.

**Review by lucacavallaro — APPROVED** [review]

**Pull request merged**